### DISCUSSION TITLE
`npx remix upgrade` command to automatically update reps

### DISCUSSION THREAD

**Discussion by mcansh** [remix-run/remix#4996] [Proposals]
found this really nice when working with storybook recently, they have `npx sb upgrade [-p]` in their CLI which will automatically find and install the latest (or prerelease) version of your installed storybook dependencies so you don't need to do anything fancy to get them all updated in lockstep (`npm i @remix-run/{node, react,..}`

## Why this would be nice

Remix deps have hard dependencies on each other and if you forget to update one or more packages, you'll end up with multiple versions installed which can cause problems, for example: https://github.com/vercel/vercel/pull/9011, #5073

```sh
Usage: sb upgrade [options]

Upgrade your Storybook packages to the latest

Options:
  --package-manager <npm|pnpm|yarn1|yarn2>  Force package manager for installing dependencies
  -N --use-npm                              Use NPM to install dependencies (deprecated)
  -y --yes                                  Skip prompting the user
  -n --dry-run                              Only check for upgrades, do not install
  -t --tag <tag>                            Upgrade to a certain npm dist-tag (e.g. next, prerelease)
  -p --prerelease                           Upgrade to the pre-release packages
  -s --skip-check                           Skip postinstall version and automigration checks
  -h, --help                                display help for command
```

PR as it only took a few minutes to do: https://github.com/remix-run/remix/pull/5004

**Comment by mjackson** [comment]
I think I'm ok with this, but not super anxious to get it done. I dislike overlapping our tools with stuff that package managers should already do for us.

Does the curly brace syntax actually work with npm already?

```
npm i @remix-run/{node,react}
```

That'd be dope if it did.

**Reply by mcansh** [reply]
it does actually! my desire for the command is mostly when upgrading to nightly or prereleases tbh

`npm i @remix-run/{node,react}@pre && npm i -D @remix-run/{dev,eslint-config}@pre` etc

**Reply by machour** [reply]
Anytime I ask someone reporting an issue the list of remix dependencies,  there's a 50% chance that they are not aligned. 

This is why I opened https://github.com/remix-run/remix/pull/5077 in the first place. 

Having a `npx remix upgrade` command as a documented way to upgrade remix will surely help.

**Reply by jsrhodes15** [reply]
Wow, I didn't know about `npm i @remix-run/{a,b,c,d}` this is 🔥

**Comment by sergiodxa** [comment]
I currently have my own `npm run upgrade:remix` that will run `npm add` and `npm add -D` for every Remix dependency I have in my package.json, so I think a built-in command can be useful.

I imagine the command could run `npm add` and `npm add -D` but with the list of dependencies/devDependencies the project has installed, it would be useful if I could specify the version like `npx remix upgrade latest` or `npx remix upgrade pre` and it can then set `@pre` to the `npm` command.

**Reply by mcansh** [reply]
these are the flags i currently have:
```log
remix upgrade (defaults to --tag latest)
remix upgrade --prerelease (same as remix upgrade --tag pre)
remix upgrade --tag nightly
```

**Comment by brophdawg11** [comment]
I've used a local script for this for some time and finally published it out the other week: https://github.com/brophdawg11/upgrade-remix

```
# defaults to `latest`
npx upgrade-remix   
npx upgrade-remix pre
npx upgrade-remix 1.16.0
```

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
